File: clase_python_01/MF0965_3_codigo_ejemplos/UF1289/Java/AuthService.java

// UF1289 - Diseño de componentes (Java)
public interface AuthService {
    boolean login(String user, String password);
    void logout(String user);
}

public class SimpleAuthService implements AuthService {
    @Override
    public boolean login(String user, String password) {
        return "admin".equals(user) && "1234".equals(password);
    }

    @Override
    public void logout(String user) {
        System.out.println(user + " logged out.");
    }
}

File: clase_python_01/MF0965_3_codigo_ejemplos/UF1289/Python/auth_service.py
# UF1289 - Diseño de componentes (Python)
from abc import ABC, abstractmethod

class AuthService(ABC):
    @abstractmethod
    def login(self, user: str, password: str) -> bool:
        pass

    @abstractmethod
    def logout(self, user: str) -> None:
        pass

class SimpleAuthService(AuthService):
    def login(self, user: str, password: str) -> bool:
        return user == "admin" and password == "1234"

    def logout(self, user: str) -> None:
        print(f"{user} logged out.")
